src/rust/rust-proto/src/protocol/service_client.rs

```diff
@@ -45,3 +51,32 @@ impl From<client_executor::Error<ConnectError>> for ConnectError {
         }
     }
 }
+
+#[async_trait::async_trait]
+pub trait ConfigConnectable
```

Comment:
Why does this need to be a separate trait? Can't this `type` just go on `Connectable`?